Add beta migration values examples and README (#373)
rstudio/helm: examples/connect/beta-migration/README.md

execution environments defined by `launcher.customRuntimeYaml` and `launcher.additionalRuntimeImages`
will be used to "bootstrap" Connect's database the first time Connect starts. Subsequent
restarts of Connect will not modify the database, even when a runtimes.yaml configuration is defined.

These sample values are for users that are evaluating the off-host execution feature
set and have content currently deployed which depends on the default set of `bionic` images.
These example values files define the default set of `bionic` images in the
`launcher.customRuntimeYaml` section, but each image's "matching" has been set to `exact`.
The `launcher.additionalRuntimeImages` contains the _new_ default set of `jammy` images,
however the `jammy` images _do not_ define a matching strategy.

Any new content that is building for the first time, or existing content that is re-building,
will attempt to match against the set of `jammy` images. Content that has already been
built with a `bionic` image can continue to _execute_ with that image until it rebuilds. If an existing
piece of content needs to be rebuilt and requires `bionic` for some reason, the Publisher

> must _explcitly_ define a default execution enviornment for that content through the content-settings

Reviewer: ```suggestion
must _explcitly_ define a default execution environment for that content through the content-settings
```

> must _explcitly_ define a default execution enviornment for that content through the content-settings

Reviewer: ```suggestion
must _explicitly_ define a default execution environment for that content through the content-settings
```

> must _explcitly_ define a default execution enviornment for that content through the content-settings

Reviewer: ```suggestion
must _explicitly_ define a default execution environment for that content through the content-settings
```

### Off-Host Execution Beta User Migration

The off-host execution beta moving to GA also moves your content from Bionic to
Jammy by default. This will cause all content to get rebuilt. If all content being
rebuilt is acceptable, then no further action is necessary.

If you prefer a phased approach to migration, you may continue using the bionic
images for existing content and jammy images for new content.

> Note: All images used by existing content should be specified in either
`launcher.customRuntimeImages` or `launcher.additionalRuntimeImages`. If content
requires both bionic and jammy images, then you must use R source packages.
Binary packages will not work when multiple OS distributions are used.

1. If you are currently using `customRuntimeImages: 'base'` (the default),
apply these changes with:

```

> helm upgrade --install -f https://raw.githubusercontent.com/rstudio/helm/main/examples/connect/beta-migration/values-base.yaml

Reviewer: Sorry, I used pseudocode on these. Still need a release name and chart reference 🙈 Happy to drop it if you don't think this is worth it. 
```suggestion
helm upgrade --install my-release rstudio/rstudio-connect --devel -f https://raw.githubusercontent.com/rstudio/helm/main/examples/connect/beta-migration/values-base.yaml
```

Author: I went ahead and removed this for now. Since these are choices that must be made during a migration I think it's best if users have to be deliberate about choosing images they want to use.

```

2. If you are currently using `customRuntimeImages: 'pro'`, apply these changes with:

```

> helm upgrade --install -f https://raw.githubusercontent.com/rstudio/helm/main/examples/connect/beta-migration/values-pro.yaml

Reviewer: ```suggestion
helm upgrade --install my-release rstudio/rstudio-connect --devel -f https://raw.githubusercontent.com/rstudio/helm/main/examples/connect/beta-migration/values-pro.yaml
```
